pkg/patterns/declarative/pkg/applier/direct_test.go

```diff
@@ -152,26 +152,34 @@ metadata:
 				return nil
 			},
 		},
-		//		{
-		//			name:      "manifest with prune and prune-whitelist",
-		//			namespace: "",
-		//			manifest: `---
-		//apiVersion: v1
-		//kind: ServiceAccount
-		//metadata:
-		//  name: foo-operator
-		//  namespace: kube-system`,
-		//			args: []string{"--prune", "--prune-whitelist=hello-world"},
-		//			expectApplyOptions: &apply.ApplyOptions{
-		//				Prune: true,
-		//			},
-		//			expectCheckFunc: func(opt *apply.ApplyOptions) error {
-		//				if opt.Prune != true {
-		//					return fmt.Errorf("prune is not set")
-		//				}
-		//				return nil
-		//			},
-		//		},
+				{
+					name:      "manifest with prune, prune-whitelist and selector",
+					namespace: "",
+					manifest: `---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: foo-operator
+  namespace: kube-system`,
+					args: []string{"--prune", "--selector", "label=true", "--prune-whitelist", "core/v1/Namespace"},
+					// TODO is this used somewhere?
+					expectApplyOptions: &apply.ApplyOptions{
+						Prune: true,
+						PruneWhitelist: []string{"core/v1/Namespace"},
+					},
+					expectCheckFunc: func(opt *apply.ApplyOptions) error {
+						if opt.Prune != true {
```

Comment:
Suggestion (non-blocking): You might try using `cmp.Diff` here; it's pretty magical ... when it works!  It's sort of like `reflect.DeepEquals` but it gives you a difference you can print in tests.  Saves you the work of doing the manual comparison.  I say "when it works" because it can fall down with complex types and require additional arguments, but this type looks simple enough that I think it will "just work".

Reply:
Cool suggestion! Haven't used it yet. I gave it a shot but it seemed like it is not the most straighforward case as `prune.PruneResource` has unexported fields. The panic message suggested to create a complex type. This would definitely be cool.

Reply:
Thanks for trying!  And yes, the unexported fields are always the problem :-(  There are some options to control this but it's a lot more complex - sorry, I thought they were "simple" all-public structs.